python/hail/dataset.py

```diff
@@ -915,6 +915,278 @@ def annotate_variants_vds(self, other, expr=None, root=None):
+                            else
+                                va.vep.most_severe_consequence
+                        """
+                    )
+                    .annotate_variants_expr(
+                        """
+                        va.gene.gene_symbol = 
+                            let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol,
+                        va.gene.gene_id =
+                            let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_id
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).gene_symbol,
+                        va.transcript_id = let canonical_transcripts = va.vep.transcript_consequences.filter(t => t.canonical == 1) in
+                            if (canonical_transcripts.exists(t => t.consequence_terms.toSet.contains(va.gene.consequence)))
+                                canonical_transcripts.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).transcript_id
+                            else
+                                va.vep.transcript_consequences.find(t => t.consequence_terms.toSet.contains(va.gene.consequence)).transcript_id
+                        """
+                    )
+                )
+
+            # drop VEP annotations if not specifically requested
+            if not veps:
+                self = self.annotate_variants_expr('va = drop(va, vep)')
+
+            # subset VEP annotations if needed
+            subset = ','.join([x for x in veps if x != 'va.vep' and x != 'va'])
+            if subset:
+                self = self.annotate_variants_expr('va.vep = select(va.vep, {})'.format(subset))
+
+        # iterate through files, selected annotations from each file
+        for file, expr in file_exprs.iteritems():
+
+            # skip VEP annotations
+            if file.startswith('gs://annotationdb/vep/'):
+                continue
+
+            # if database file is a VDS
+            if file.endswith('.vds'):
+
+                # annotate analysis VDS with database VDS
+                self = self.annotate_variants_vds(self.hc.read('{}'.format(file)), expr = expr)
```

Comment:
For some reason, it doesn't like being passed the variable `file`. I get:
```
hail.java.FatalError: An error occurred while calling into JVM, probably due to invalid parameter types.

Java stack trace:
An error occurred while calling o54.arrayListToISeq. Trace:
py4j.Py4JException: Method arrayListToISeq([class java.lang.String]) does not exist
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:318)
	at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:326)
	at py4j.Gateway.invoke(Gateway.java:272)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)


Hail version: 0.1-9a44a31
Error summary: An error occurred while calling into JVM, probably due to invalid parameter types.
```
Doesn't happen when I pass the formatted string.